onnxruntime/core/util/thread_utils.cc

```diff
@@ -13,55 +13,81 @@ static inline std::vector<size_t> GenerateVectorOfN(size_t n) {
   std::vector<size_t> ret(n);
   std::iota(ret.begin(), ret.end(), 0);
   return ret;
-                }
+}
+
+struct NumCoresUtil {
 #ifdef _WIN32
-		// This function doesn't support systems with more than 64 logical processors
-		static std::vector<size_t> GetNumCpuCores() {
-			// Indeed 64 should be enough. However, it's harmless to have a little more.
-			SYSTEM_LOGICAL_PROCESSOR_INFORMATION buffer[256];
-			DWORD returnLength = sizeof(buffer);
-			if (GetLogicalProcessorInformation(buffer, &returnLength) == FALSE) {
-                          return GenerateVectorOfN(std::thread::hardware_concurrency());
-                        }
-                        std::vector<size_t> ret;
-			int count = (int)(returnLength / sizeof(SYSTEM_LOGICAL_PROCESSOR_INFORMATION));
-			for (int i = 0; i != count; ++i) {
-				if (buffer[i].Relationship == RelationProcessorCore) {
-					ret.push_back(buffer[i].ProcessorMask);
-				}
-			}
-			if (ret.empty())
-                          return GenerateVectorOfN(std::thread::hardware_concurrency());
-                        return ret;
-                }
+  // This function doesn't support systems with more than 64 logical processors
+  static std::vector<size_t> GetNumCpuCores() {
```

Comment:
Unlike WindowsEnv::GetNumCpuCores(), this function returns a list of affinity mask of logical processor in current processor group for Windows. Change the function name to GetAffinityMasks()?

Reply:
Sure, will do.